tools/github-action-reporter.js

```diff
@@ -0,0 +1,79 @@
+module.exports = {
+  "reporter:respec-github-action": ["type", GithubActionReporter],
+};
+
+function GithubActionReporter() {
+  const browserFailures = [];
+  this.onSpecComplete = (_browser, result) => {
+    if (!result.success) {
+      const failure = parseFailure(result);
+      browserFailures.push(failure);
+    }
+  };
+
+  this.onRunComplete = () => {
+    print("group", "Failed tests");
+    browserFailures.forEach(printFailure);
+    print("endgroup");
+  };
+}
+
+/**
+ * @typedef {{ test: string[], file: string, line: number, col: number, message: string }} Failure
+ * @returns {Failure}
+ */
+function parseFailure(result) {
+  // convert newlines into array and flatten
+  const log = result.log.flatMap(message => message.split("\n"));
+  const { suite, description } = result;
+  const message = log[0].replace("Error: ", "");
+  const location = log[2].split(":9876/base/", 2)[1].replace(/\)$/, "");
```

Comment:
Could you extract the port number as a const with some comment? Might be easier to understand later.